adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -539,6 +639,70 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Prepares two RDDs to be joined with any shuffleRegionJoin. This includes copartition
+   * and sort of the rightRdd if necessary.
+   *
+   *  @param genomicRdd The RDD to join to.
+   *  @return A tuple containing both RDDs ready to join, the partitionMap and the number
```

Comment:
If we're returning a tuple of 4 things, it might be worth turning this into a case class instead. CC @heuermh for thoughts.